portal/server/controllers/dashboard/filters/generate-selected-filters.js

```diff
@@ -35,18 +34,59 @@ const generateSelectedFiltersObject = (
  * Create an object for all selected submissionType filters.
  * This will used in mojFilter component - selectedFilters.categories.
  *
+ * @param {string} field name
  * @param {object} submitted submissionType filters
  * @example ( ['Automatic Inclusion Notice', 'Manual Inclusion Notice'] )
  * @returns generateSelectedFiltersObject('Notice Type', 'submissionType', ['Automatic Inclusion Notice', 'Manual Inclusion Notice'])
  */
-const selectedSubmissionTypeFilters = (submittedFilters) =>
+const selectedSubmissionTypeFilters = (fieldName, submittedFilters) =>
   generateSelectedFiltersObject(
     CONTENT_STRINGS.DASHBOARD_FILTERS.FILTER_HEADINGS.NOTICE_TYPE,
-    CONSTANTS.FIELD_NAMES.DEAL.SUBMISSION_TYPE,
+    fieldName,
     submittedFilters,
   );
 
+/**
+ * Map true/false string to Issued/Unissued string.
+ *
+ * @param {string} the submitted filter value
+ * @example ( 'true' )
+ * @returns 'Issued'
+ */
+const mapIssuedValueToText = (hasBeenIssued) => {
+  if (hasBeenIssued === 'true') {
```

Comment:
should this be a boolean value?

Reply:
Not with how it works at the moment - I know it seems strange; what happens is:

1) Generate filters to display in the page (with expected boolean format) e.g:

```js
{
  type: [
    { text: 'Loan', value: 'Loan' },
    { text: 'Cash' value: 'Cash' }
  ],
{
  hasBeenIssued [
    { text: 'Issued', value: true},
    { text: 'Unissued' value: false }
  ],
}
```
2) Nunjucks renders elements with this - Nunjucks converts boolean values to strings
3) User submits some filters, UI server receives these in `req.body`
4) Submitted filters are added to `req.session`- at this point `hasBeenIssued` values are strings.
5) Query to API (as part of this, string booleans are converted to booleans for 'correct' DB query)
6) Re-render the page with selected filters - the selected filters are sourced from the `req.session`, rather than re-mapping with the sent DB query that has actual booleans


Hope it makes sense! In next PR I need to change and improve how the filters are sent to the API, to fix a bug apart from anything else. Maybe as part of this I can also look at immediately converting any string booleans in the submitted filters, before it's added to the session. Or could do as part of this PR if you'd like?

Reply:
next PR is fine, no problemo :)